subworkflows/nf-core/bam_dedup_umi/main.nf

```diff
@@ -0,0 +1,131 @@
+//
+// BAM deduplication with UMI processing
+//
+
+include { BAM_DEDUP_STATS_SAMTOOLS_UMICOLLAPSE as BAM_DEDUP_STATS_SAMTOOLS_UMICOLLAPSE_TRANSCRIPTOME } from '../../../subworkflows/nf-core/bam_dedup_stats_samtools_umicollapse'
+include { BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS as BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS_TRANSCRIPTOME       } from '../../../subworkflows/nf-core/bam_dedup_stats_samtools_umitools'
+include { BAM_DEDUP_STATS_SAMTOOLS_UMICOLLAPSE as BAM_DEDUP_STATS_SAMTOOLS_UMICOLLAPSE_GENOME        } from '../../../subworkflows/nf-core/bam_dedup_stats_samtools_umicollapse'
+include { BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS as BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS_GENOME              } from '../../../subworkflows/nf-core/bam_dedup_stats_samtools_umitools'
+include { BAM_SORT_STATS_SAMTOOLS                            } from '../../../subworkflows/nf-core/bam_sort_stats_samtools'
+include { UMITOOLS_PREPAREFORRSEM                            } from '../../../modules/nf-core/umitools/prepareforrsem'
+include { SAMTOOLS_SORT                                      } from '../../../modules/nf-core/samtools/sort/main'
+
+workflow BAM_DEDUP_UMI {
+    take:
+    ch_genome_bam         // channel: [ val(meta), path(bam), path(bai) ]
+    ch_fasta              // channel: [ val(meta), path(fasta) ]
+    umi_dedup_tool        // string: 'umicollapse' or 'umitools'
+    umitools_dedup_stats  // boolean: whether to generate UMI-tools dedup stats
+    bam_csi_index         // boolean: whether to generate CSI index
+    ch_transcriptome_bam  // channel: [ val(meta), path(bam) ]
+    ch_transcript_fasta   // channel: [ val(meta), path(fasta) ]
```

Comment:
I would have put all the channels first (references and then files), and then the string and boolean.
But that's more or a me thing I believe than a proper convention